dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/citation/StandardCitationFormatsConverter.java

```diff
@@ -46,9 +46,21 @@ public String toString(CitationData data, Locale locale, boolean escapeHtml) {
                 .map(GlobalId::toURL)
                 .map(URL::toString)
                 .orElse(StringUtils.EMPTY);
-        citation.urlValue(pid, pid).endPart()
-                .value(data.getPublisher()).endPart()
-                .rawValue(data.getVersion()).endPartEmpty();
+        citation.urlValue(pid, pid);
+
+        if(data.getPublisher() == null && data.getVersion() == null) {
+            citation.endPartEmpty();
+        } else {
+            citation.endPart();
+        }
+
+        if(data.getPublisher() != null) {
+            citation.value(data.getPublisher()).endPart();
+        }
+        if(data.getVersion() != null) {
+            citation.rawValue(data.getVersion()).endPartEmpty();
+        }
```

Comment:
Probably it would be a little easier to understand what's going on if we would switch logic here a little: Defining what should be before each part instead of what should be at the end:
```
citation.urlValue(pid, pid).endPartEmpty()
    .add(", ").value(data.getPublisher()).endPartEmpty()
    .add(", ").rawValue(data.getVersion()).endPartEmpty()
```

Reply:
Didn't know that `,` would not be added if value was missing.